### PR TITLE
[REVIEW] JNI and Java support for is_nan and is_not_nan

### PR DESCRIPTION
**Pull request by kuhushukla**
Add JNI bindings and java api for is_nan and is_not_nan. Currently needs the compile fix per : https://github.com/rapidsai/cudf/pull/4037 to compile and test.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -779,6 +779,30 @@ public ColumnVector isNull() {
     }
   }
 
+  /**
+   * Returns a Boolean vector with the same number of rows as this instance, that has
+   * TRUE for any entry that is Nan, and FALSE otherwise
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: What happens if the input value is null?

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 2]
```diff
@@ -779,6 +779,30 @@ public ColumnVector isNull() {
...
+   */
+  public ColumnVector isNan() {
+    try (DevicePrediction prediction = new DevicePrediction(predictSizeFor(DType.BOOL8),"isNan")) {
+      return new ColumnVector(isNanNative(offHeap.cudfColumnHandle.getViewHandle()));
+    }
+  }
+
+  /**
+   * Returns a Boolean vector with the same number of rows as this instance, that has
+   * TRUE for any entry that is not Nan, and FALSE otherwise
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Here too what happens is the input value is null?  What about if it is not a double or float?

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 3]
```diff
@@ -228,6 +228,78 @@ void isNullTestEmptyColumn() {
     }
   }
 
+  @Test
+  void isNanTest() {
+    try (ColumnVector v = ColumnVector.fromBoxedDoubles(1.0, 2.0, Double.NaN, 4.0, Double.NaN, 6.0);
```
Comment (revans2) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: Can we have some tests that include floats not just doubles and some nulls as well?

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java (hunk 1 above)
Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: Will check and add tests.

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java (hunk 3 above)
Comment (kuhushukla) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: You got it! Coming up in the next round.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java (hunk 1 above)
Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java]: nulls are not nans is the behavior based on added test

**Comment by revans2** [comment]
please rerun tests

**Comment by revans2** [comment]
please rerun tests

**Pull request merged**